script.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+// form issue 해결
+// submit이벤트가 발생하면 브라우저가refresh된다.
+// e.preventDefault()를 통해서 브라우저의 refresh를 막고
+// 새로운 변수 location에 input값 저장해서 textContent로 출력
+const Form = document.querySelector("form");
+const inputBox = document.querySelector("input");
+const waitingList = document.querySelector("#waitingList");
+
+Form.addEventListener("submit", (e) => {
+  e.preventDefault();
+  //const location = inputBox.value;
+  //waitingList.textContent = location;
+});
+
+// input창에 입력
+let itemList = [];
+let finishedItem = [];
+
+let inputButton = document.querySelector("button");
+inputButton.addEventListener("click", addItem);
+
+function addItem() {
+  let item = document.querySelector("#input").value;
+  if (item != null) {
+    itemList.push(item);
+    document.querySelector("#input").value = "";
+    document.querySelector("#input").focus();
+  }
+  showList();
+}
+
+// ul출력
+
+function showList() {
+  let list = "";
+  for (let i = 0; i < itemList.length; i++) {
+    list +=
+      "<li>" +
+      itemList[i] +
+      "<button class='close' id=" +
+      i +
+      ">" +
+      "<img src='img/bin.png'/>" +
+      "</button></li>";
+    // list +=
+    //   "<li id='list" +
+    //   i +
+    //   "'><button class='listBtn' onclick='movingFinish(" +
+    //   i +
+    //   ")'>" +
+    //   "<span class='text'>" +
+    //   itemList[i] +
+    //   "</span></button><button class='close' id=" +
+    //   i +
+    //   ">" +
+    //   "<img src='img/bin.png'/>" +
+    //   "</button></li>";
+  }
+  document.getElementById("waitingList").innerHTML = list;
+  console.log(itemList);
+  waitingCount();
+
+  let deleteButtons = document.querySelectorAll(".close");
+  for (let i = 0; i, deleteButtons.length; i++) {
+    deleteButtons[i].addEventListener("click", deleteItem);
+  }
+}
+
+function deleteItem() {
+  let id = this.getAttribute("id");
+  itemList.splice(id, 1);
+  showList();
+}
+
+// 대기중의 숫자 count
+
+function waitingCount() {
```

Comment:
```suggestion
function countWaitingItems() {
```

```suggestion
function renderWaitingItemsCount() {
// waiting items의 길이를 가져오고
// 그 길이 숫자를 화면에 렌더링한다!
```

- 함수는 동작을 나타내기 때문에 함수의 이름은 동사로 (시작)합니다!
- 함수의 이름은 함수가 하는 일을 충분히 구체적으로 자세하게, 모호하지 않게 지어주세요.

Reply:
헉.. 구체적인 답변 정말 감사합니다! 제가 몰랐던 부분 많이 알아갈 수 있었습니다!
참고해서 리펙토링 진행하도록 하겠습니다!